src/libraries/Microsoft.NETCore.Platforms/pkg/runtimeGroups.props

```diff
@@ -91,6 +91,17 @@
       <Versions>8;9;10;11;12;13</Versions>
     </RuntimeGroup>
 
+    <RuntimeGroup Include="ios-sim">
```

Comment:
Just realized this will be the first RID that uses `-` in the base portion.  This is used as a delimiter elsewhere.  I wonder if this will cause any issues with folks trying to parse RIDs. 

For example, this will be broken:
https://github.com/dotnet/runtime/blob/1bf0610e30dff8b3fb1328a2a8772e738d73157a/Directory.Build.props#L111

Reply:
Are you proposing fixing cases that use IndexOf to parse, or changing the RID to something with no dash?

Reply:
I'm highlighting a concern, I haven't looked for all places that might do this, nor am I sure how to find all those.  I recall the SDK did this parsing at one point.  @vitek-karas might have a better idea.  If it's only in our infra we could change it, if not than perhaps we should select a different pattern.  

What made you choose `-sim`, does ios/tvos expose this?  Is there any natural choice that's in line with what our PAL would discover from the system? https://github.com/dotnet/runtime/blob/59d6f74a708068584d96177e7cbf6da41dcf1dbd/src/native/corehost/hostmisc/pal.unix.cpp#L534-L796

Reply:
I don't know in SDK, in the host we don't parse the RIDs in any way. But there's a desire to start doing that (since the RID fallback graph is getting out of hand and we have to serialize it into .deps.json for the host to work, and it breaks on new distros since we have to add them into the graph for things to work). I would definitely prefer if the parsing logic could be reasonably simple for the host to have it. The plan there is that the host will be able to imply certain fallbacks without them being written in the graph explicitly.
I think the version fallbacks is probably the one we want to do the most - that is automatically fallback from `ios.10-arm` to `ios-arm` (for example). I guess this could work if we change it like `ios.10-sim-x86` which would automatically fall back to `ios-sim-x86`.
The other fallback which should probably be automatic is simply dropping the architecture, so `ios-arm` falls back to `ios` automatically. But this is probably bit more tricky (due to things like `win-aot` and `win-x86-aot`).

I definitely think that we should follow the existing scheme for versions which is `id.##-...` adding the dash before the version doesn't feel right, it breaks the existing pattern.

Reply:
So, quick vote, what is the preferred vision for how "iOS 10 x86 simulator" RID looks?

Currently, the way runtimeGroups.props gets converted to runtime.json and runtime.compatibility.json means the answer is "ios-sim.10-x86"

Changing the RID to something with no dash would result in e.g. iossimulator-10.x86. Is that preferred? Rewriting the RID generator to somehow do ios.10-sim-x86 is also an option.

What would people prefer, here?

Reply:
If I'm following correctly, we should change the name to something like `iossim` or `iossimulator` instead of `ios-sim`.

Reply:
I think `iossim` would work (or similar). Please note that `iossimulator-10.x86` is also wrong. The pattern should be `iossimulator.10-x86` (maybe it was just a typo).

Reply:
Out of curiosity - if I'm on Apple M1 (arm64) and I'm creating an iPhone app - it will need to be build separately for simulator and separately for the device? (even if it's the same arch and everything).

Reply:
And one more: is iOS simulator a superset of iOS? That's how the RIDs are defined right now.

Reply:
>  it will need to be build separately for simulator and separately for the device? 

correct and you also need to pull different native nuget dependencies

Reply:
@ericstj does it make sense to you to change the name?

Reply:
Yeah, I think so unless there is some other constraint that's forcing us to use the `-` (like that's what uname returns or something).  We've been trying to follow some patterns to add structure to the RIDs even though there's nothing enforced ATM.  The current structure is that `.` is the delimiter between base portion and version (except for `win`) and `-` is the delimiter between version and architecture.